### PR TITLE
BoardElections && CSS improvements

### PR DESCRIPTION
**Pull request by olblak**
I adapted the nice work done by @ndeloof from PR [#96](https://github.com/jenkins-infra/account-app/pull/96) and made some fixes and improvements.

Bugs:
* Within Boardelection form, drag&drop wasn't working anymore because of Jquery broken links.
  -> Jquery is now packages with the application (no external links)
* Variable 'electionLogfile' wasn't taken into account
  -> I changed electionLogfile to electionLogdir as the account app only need to know where to create election log files
* All votes where stored in a simple text file which had several limitations.
  -> I am now storing votes in a yaml file, with one yaml file per election
* Change 'commons-jelly-tags-define' library
  -> commons-jelly:commons-jelly-tags-define:1.0
  I still don't know why but I had several issues with org.jvnet.hudson:commons-jelly-tags-define
  from Jetty docker container (but not 'locally').

Improvement (UI && CSS):
* I added a sidebar menu to navigate to Board Election, Admin, Profile Update
* I added css rules for some forms (Login, Sign in, Reset Password, Update Profile)
* Only show sidebar menu when users are logged
* Add a Makefile to simply clean/build/run actions

What's still broken:
* Several external links from footer/menu are still broken
* Missing icons

Remarks:
I have no idea if it's a good idea or but I am fetching four libraries from maven-central
* com.esotericsoftware.yamlbeans:yamlbeans:1.11
* org.webjars:webjars-servlet-2.x:1.5
* org.webjars:jquery:3.2.0
* org.webjars:jquery-ui:1.12.1

In order to test and validate this PR:

1. Create file '.env' next to docker-compose.yaml with content :
```
    APP_URL=http://localhost:8080/
    ELECTION_CANDIDATES=alice,bob
    ELECTION_CLOSE=2038/01/19
    ELECTION_OPEN=1970/01/01
    JIRA_USERNAME=<insert your jira username>
    JIRA_PASSWORD=<insert your jira password>
    JIRA_URL=https://issues.jenkins-ci.org
    LDAP_URL=server=ldap://localhost:9389/
    LDAP_PASSWORD=<insert your ldap password>
    LDAP_MANAGER_DN=cn=admin,dc=jenkins-ci,dc=org
    LDAP_NEW_USER_BASE_DN=ou=people,dc=jenkins-ci,dc=org
    RECAPTCHA_PRIVATE_KEY=recaptcha_private_key
    RECAPTCHA_PUBLIC_KEY=recaptcha_public_key
```

2. Run ```ssh -4 -L 9389:localhost:389 ldap.jenkins.io```
  In another terminal
3. Run ```make clean build run```
4. Then go to http://localhost:8081
5. As soon as a vote is done, a new file will appear in /var/log/accountapp/elections 
```docker exec -i -t accountapp_run_1 cat /var/log/accountapp/elections/20380101.yaml```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,15 @@
+.PHONY: build clean run
+
+current_dir := $(shell pwd)
+current_user := $(shell id -u)
+
+build:
+	docker run --rm -v $(current_dir):/opt/accountapp/ -u $(current_user):$(current_user) -w /opt/accountapp --entrypoint ./gradlew java:8 --no-daemon --info war
```
Comment (rtyler) [inline Makefile]: While I find this approach weird (who doesn't have JDK8 locally now :smile:), the java container is deprecated, please rely on the `openjdk` container instead.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+	docker-compose build run
+
+clean:
+	docker run --rm -v $(current_dir):/opt/accountapp/ -u $(current_user):$(current_user) -w /opt/accountapp --entrypoint ./gradlew java:8 clean
+	docker-compose rm -f run
+
+run:
```
Comment (rtyler) [inline Makefile]: shouldn't this depend on the `build` target?

File: build.gradle [hunk 3]
```diff
@@ -12,6 +12,7 @@ repositories {
     jcenter()
     maven { url 'https://repo.jenkins-ci.org/public' }
     mavenLocal()
+    mavenCentral()
```
Comment (rtyler) [inline build.gradle]: This isn't necessary, `jcenter()` acts as a Maven central cache and provides a number of other artifacts not published to maven central.

**Comment by olblak** [comment]
Each time someone made a vote, the result was added to "$election_logfile" which created a lot of duplications.
Let's say we have three users Alice, Bob and John

The way it was working before:
  2017/01/01: John decided to vote for Alice&Bob
    -> The vote added 'John: Alice, Bob' to the file "$election_logfile"
  2017/01/02: John change his mind and decide to vote for Bob only
    -> The vote added 'John: Bob' to the file $election_logfile
That means that $election_logfile contain
---
John: Alice, Bob
...
John: Alice
---

So if we want to be sure that we only have one entry per user, we must parse $election_logfile to see if someone already voted or note.
  
With a yaml file it becomes easier, 
  1) We open the yaml file
  2) We add entry 'user_id:vote', which replace previous 'user_id'.
  3) We write the results to the yaml file

File: Makefile (hunk 1 above)
Comment (olblak) [inline Makefile]: I suppose those who don't have ruby locally :)
Joke aside, I just reused the same docker image that the one defined in Jenkinsfile.
So we are sure to have the 'same' container that we one build by Jenkins
So I also update the Jenkinsfile to use openjdk:8-jdk

File: Makefile (hunk 2 above)
Comment (olblak) [inline Makefile]: Personally, I prefer to keep them separated and run ```make build run```  so I clearly know what I am doing

File: build.gradle (hunk 3 above)
Comment (olblak) [inline build.gradle]: Good to know, I remove mavenCentral()

**Comment by olblak** [comment]
I just fixed fontAwesome icon link in BoardElection form

**Pull request merged**